opensearch/src/main/java/org/opensearch/sql/opensearch/response/OpenSearchResponse.java

```diff
@@ -110,7 +109,14 @@ public Iterator<ExprValue> iterator() {
       return exprValueFactory.getParser().parse(aggregations).stream().map(entry -> {
         ImmutableMap.Builder<String, ExprValue> builder = new ImmutableMap.Builder<>();
         for (Map.Entry<String, Object> value : entry.entrySet()) {
-          builder.put(value.getKey(), exprValueFactory.construct(value.getKey(), value.getValue()));
+          builder.put(
```

Comment:
Maybe it is offtopic here, but can you split this huge function into multiple ones?

Reply:
Sure thing.